ts/javascript/initializers.ts

```diff
@@ -0,0 +1,207 @@
+import { Initializer } from "../base.ts";
+import * as fs from "https://deno.land/std@0.201.0/fs/mod.ts";
+import { assertEquals } from "https://deno.land/std@0.201.0/assert/mod.ts";
+import outdent from "https://deno.land/x/outdent@v0.8.0/mod.ts";
+import { mapValues } from "../internal/utils.ts";
+
+const npmInitializer: Initializer = () => {
+  const existsPkgJson = fs.existsSync("package.json");
+  const existsPkgLock = fs.existsSync("package-lock.json");
+  const existsYarnLock = fs.existsSync("yarn.lock");
+  if (!existsPkgJson || (!existsPkgLock && existsYarnLock)) {
+    return { tag: "ShouldNotRun" };
+  }
+  const contents = Deno.readTextFileSync("package.json");
+  try {
+    const packageJson = JSON.parse(contents);
+    const scripts = mapValues(
+      (script: string) => script.replaceAll("`", "\\`"),
+      packageJson.scripts ?? {}
+    );
+    const testScript: string | undefined = scripts.test;
+    const otherScripts = Object.entries(scripts).filter(
+      ([name]) => name !== "test"
+    );
+    return {
+      tag: "ShouldRun",
+      imports: [],
+      makeTarget: () =>
+        [
+          outdent`
+          export const ${packageJson.name || "npmProject"} = mkNpmProject({
+            description: "${packageJson.description || "An NPM project"}",
+            src: ".",
+            nodeVersion: "18",
+          })`,
```

Comment:
```suggestion
          outdent`
            export const ${packageJson.name || "npmProject"} = mkNpmProject({
              description: "${packageJson.description || "An NPM project"}",
              src: ".",
              nodeVersion: "18",
            })`,
```

Reply:
Does `outdent` remove trailing spaces? If yes, I would add a newline before the backtick.